platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -174,22 +168,20 @@ - (void)setURL:(NSURL *)URL
     self.rawSource->setURL(url.absoluteString.UTF8String);
 }
 
-- (void)setFeatures:(NSArray *)features
+
+- (void)setShape:(MGLShape *)shape
 {
     if (self.rawSource == NULL)
     {
         [self commonInit];
     }
-    
-    mbgl::FeatureCollection featureCollection;
-    featureCollection.reserve(features.count);
-    for (id <MGLFeaturePrivate> feature in features) {
-        featureCollection.push_back([feature mbglFeature]);
-    }
-    const auto geojson = mbgl::GeoJSON{featureCollection};
+
+    const auto geojson = mbgl::GeoJSON{shape.geometryObject};
     self.rawSource->setGeoJSON(geojson);
     
-    _features = MGLFeaturesFromMBGLFeatures(featureCollection);
+    [self willChangeValueForKey:@"shape"];
```

Comment:
These change notifications are unnecessary. Since setShape: is a standard KVC selector, these change notifications will be generated for you implicitly.